analysis/staticAnalysisReport.Rmd

```diff
@@ -0,0 +1,171 @@
+#knitr::kable(head(arrange(outlyingRows, desc(freq)) , 5) , col.names = c("Package" , "Number of evals"))
+#lineCountPerPkg <- 
+#  lineCountPerFile %>%
+#    group_by(pkg) %>%
+#      summarise(lineCount = sum(lineCount), charCount = sum(charCount))
+#lineNdEvalCountPerPkg <- merge(lineCountPerPkg, evalCntPerPkg)
+#evalCntPerPkgLeft <- evalCntPerPkg[ !(evalCntPerPkg$pkg %in% lineCountPerPkg$pkg) , ]
+#evalCntPerPkgLeft
+```
+
+## Load the data
+Usage of the function eval in R-packages.
+To get evals.csv, run - Rscript evalCrawler.R /mnt/nvme0/R/CRAN-extracted
+```{r}
+evaldf <- read_csv("evals.csv")
+```
+
+## Plot occurences of evals in packages
+```{r}
+ggplot(evaldf , aes(pkg)) + 
+  geom_bar() + 
+  theme(axis.text.x = element_blank()) + 
+  labs(y="Number of evals", x="packages", title="Figurative distribution of the usage of eval in various packages")
+```
+
+Top 20 packages using eval
+```{r}
+evalCntPerPkg <- dplyr::count(evaldf, pkg)
+knitr::kable(
+  evalCntPerPkg %>%
+  top_n(20, wt=n) %>%
+  arrange(desc(n)))
+```
+
+Eval distribution in packages - density and histogram plots
+```{r}
+ggplot(evalCntPerPkg, aes(n)) + 
+  geom_density()+ scale_x_log10() + 
+  labs(y="Number of projects (density)", x="Number of Evals - (log scaled)", title="Evals distribution in the packages")
+ggplot(evalCntPerPkg , aes(n)) + 
+  geom_histogram(bins=50) + 
+  scale_x_log10() + 
+  labs(y="Number of projects", x="Number of Evals", title="Evals distribution in the packages - histogram")
+```
+
+Exploring outliers
+```{r}
+ggplot(evalCntPerPkg , aes(x = "" , y = n)) +
+  geom_boxplot() +
+  labs(y="Number of Evals", x="Packages", title="Evals distribution in the packages - boxplot")
+```
+
+```{r}
+outlyingVals <- boxplot.stats(evalCntPerPkg$n)$out
+outlyingRows <- filter(evalCntPerPkg, n %in% outlyingVals)
+evalCountPerFile <- dplyr::count(evaldf, path)
+```
+
+Top Outliers
+```{r}
+##top outliers
+knitr::kable(
+  evalCountPerFile %>% 
+    top_n(5, wt=n) %>% 
+    arrange(desc(n)))
+```
+
+Bottom Outliers
+```{r}
+knitr::kable(
+  evalCountPerFile %>% 
+    top_n(5, wt=n) %>% 
+    arrange(n))
+#knitr::kable(head(arrange(outlyingRows, n) , 5) , col.names = c("Package" , "Number of evals"))
+```
+
+Plotting without outliers
+```{r}
+evalCntPerPkgMod <- evalCntPerPkg[ !(evalCntPerPkg$n %in% outlyingVals), ]
+ggplot(evalCntPerPkgMod , aes(x = "" , y = n)) +
+  geom_boxplot() +
+  labs(y="Number of Evals", x="Packages", title="Evals distribution in the packages - boxplot (outliers removed)")
+```
+
+##Some interesting plots 
+
+```{r}
+knitr::kable(tribble(
+  ~A,~B,
+  "Total Number of eval occurences combined" , nrow(evaldf),
+  "Total Number of packages considered" , "12574",  #the latter is from the output of the evalCrawler.R script
+  "Number of packages using eval" , nrow(unique.data.frame(evaldf["pkg"]))
+))
+```
+
+Distribution as per usage
+```{r}
+# Eval occurences per use
```

Comment:
What is the difference between: "Distribution as per usage" and "Eval occurences per use"?

Reply:
Removed it.